db/silkworm/stagedsync/stage_execution.cpp

```diff
@@ -98,4 +102,117 @@ StageResult stage_execution(lmdb::DatabaseConfig db_config) {
 
     return StageResult::kStageSuccess;
 }
+
+void collect_for_unwind(Bytes key, Bytes value, lmdb::Table* plain_state_table, lmdb::Table* plain_code_table) {
+    if (key.size() == 20) {
+        plain_state_table->del(key);
+        if (value.size() > 0) {
+            auto address{key.substr(0, kAddressLength)};
+            auto [account, err]{decode_account_from_storage(value)};
+            rlp::err_handler(err);
+            if (account.incarnation > 0 && account.code_hash == kEmptyHash) {
+                Bytes code_hash_key(kAddressLength + db::kIncarnationLength, '\0');
+                std::memcpy(&code_hash_key[0], &address[0], kAddressLength);
+                boost::endian::store_big_u64(&code_hash_key[kAddressLength], account.incarnation);
+                auto new_code_hash{*plain_code_table->get(code_hash_key)};
+                std::memcpy(&account.code_hash.bytes[0], &new_code_hash[0], kHashLength);
+            }
+            // cleaning up contract codes
+            auto state_account_encoded{plain_state_table->get(address)};
+            if (state_account_encoded != std::nullopt) {
+                auto [state_incarnation, err]{extract_incarnation(*state_account_encoded)};
+                rlp::err_handler(err);
+                // cleanup each code incarnation
+                for (uint64_t i = state_incarnation; i > account.incarnation && i > 0; --i) {
+                    Bytes key_hash(kAddressLength + 8,'\0');
+                    std::memcpy(&key_hash[0], key.data(), kAddressLength);
+                    boost::endian::store_big_u64(&key_hash[kAddressLength], i);
+                    plain_code_table->del(key_hash);
+                }
+            }
+
+            auto new_encoded_account{account.encode_for_storage(false)};
+            plain_state_table->put(key, new_encoded_account);
+        }
+        return;
+    }
+    
+    auto location{key.substr(kAddressLength + db::kIncarnationLength)};
+    plain_state_table->del(key.substr(0, kAddressLength + db::kIncarnationLength), location);
+    if (value.size() > 0) {
+        auto data{location.append(value)};
+        plain_state_table->put(key.substr(0, kAddressLength + db::kIncarnationLength), data);
+    }
+    return;
+}
+
+void walk_collect(lmdb::Table* source, lmdb::Table* plain_state_table, lmdb::Table* plain_code_table, uint64_t block_number, uint64_t unwind_to) {
+    Bytes unwind_to_bytes(8, '\0');
+    boost::endian::store_big_u64(&unwind_to_bytes[0], unwind_to+1);
+    MDB_val mdb_key{db::to_mdb_val(unwind_to_bytes)};
+    MDB_val mdb_data;
+    int rc{source->seek(&mdb_key, &mdb_data)}; 
+    while (!rc) {
```

Comment:
Does not look right to me. When `rc != 0` (apart from `MDB_NOTFOUND`), it is an error and we want to break the loop instead of continuing.